### DISCUSSION TITLE
Thumbnail disk space usage

### DISCUSSION THREAD

**Discussion by djjudas21** [photoprism/photoprism#2129] [Q&A] [answered]
I'm in the process of trialling PhotoPrism and so I've imported a portion of my photo library to test it out. I imported ~53GB of photos with default settings, and the space required by the thumbnails is ~18GB.

I wasn't happy with that usage so I set 

```
PHOTOPRISM_THUMB_SIZE: "1920"
PHOTOPRISM_JPEG_QUALITY: "85"
```

and regenerated them. The thumbnails now take ~13GB, which is an improvement but I don't like the roughly 1:3 ratio of thumbnail size to image size, especially as the photo library I want to eventually import is about 600GB.

Is there anything else I can do to make it generate either smaller or fewer thumbnails? I notice by default it generates 11 thumbs for each image.

**Comment by lastzero** [comment, marked as answer]
You can further compress them, I think 80 is what many hosted apps would use. Personally, I care for better quality. Note you will rarely view originals in a browser and browsers typically also do a bad job with downsizing. So thumb quality is as important as the quality of your originals in many cases.

**Reply by djjudas21** [reply]
Thanks for the advice. What about the number of thumbnails? By reducing the static size to 1920 from 2048 it now generates 10 thumbnails per original rather than 11.

The PC I typically use for browsing photos only has 1920x1080 resolution anyway.

**Reply by graciousgrey** [reply]
Here's described which thumbnails are generated by PhotoPrism for what purpose: https://docs.photoprism.app/user-guide/settings/advanced/#what-files-will-be-created-by-photoprism. In case you don't need the big ones you can reduce the static size limit. Note that changing the thumbnail settings won't delete existing thumbnails and that face detection accuracy will be reduced.

**Reply by LostSoulfly** [reply]
Is there perhaps an option or planned feature to use the original image for face detection and recognition, or perhaps generating then deleting (or not even writing to disk), eliminating the need for so many high quality thumbnails? I'm perfectly fine with indexing taking longer -- it's not something that will be done often -- but I really find it hard to swallow such a large cache folder but do not want to sacrifice face detection/recognition quality.

Alternatively, while the documentation has a lot of information, it would be very nice to know what can be safely reduced so as to not impact detection/recognition. It's just not clear what PHOTOPRISM_JPEG_SIZE does vs PHOTOPRISM_THUMB_SIZE_UNCACHED.

From [this comment](https://github.com/photoprism/photoprism/issues/1647#issuecomment-948912335), it seems that I HAVE to generate and store large thumbnails to make effective use of the facial detection/recognition, even with PHOTOPRISM_THUMB_SIZE_UNCACHED set high. But it seems like that is an effective alternative provided you have enough power. But then he says that PHOTOPRISM_JPEG_SIZE is too low which would cause it to fail anyway? But apparently PHOTOPRISM_JPEG_SIZE appears to be only related to RAW files, which should probably be stated in the [Config Options page](https://docs.photoprism.app/getting-started/config-options/) for clarity, but then that makes no sense. So I'm completely lost here. Additionally, I have no RAW files so if my understanding is correct, it should still work, no?

**Reply by djjudas21** [reply]
Just for a data point, running with the settings I was getting a ratio of approx 3:1 between the size of the originals and the size of the thumbnails.

I then set the following and started over:

```
PHOTOPRISM_THUMB_SIZE: "1920"
PHOTOPRISM_JPEG_QUALITY: "85"
```

... and I'm now getting a ratio of more like 10:1. I haven't noticed any problems with facial recognition (as in, it still works) but I haven't tested it for accuracy or whether it can detect some of the smaller faces in the scene. I guess the ratio depends on the size and quality of your originals. Mine are mostly iPhone 7 JPGs (12 MP) with a smattering of 18MP JPGs from a Canon 600D.

I think @BradleyShaner's suggestion is good. Could the facial recognition step be done on the original at import time, rather than on a thumbnail? Or at least on a higher-res thumbnail that is then not kept?

In my specific case, I don't really have a use case for thumbnails greater than 1920px (none of my devices have screens bigger than this) but the facial recognition is still important. And of course I can reconfigure and regenerate the thumbnails at a larger size if Santa brings me a 4K screen!

**Reply by lastzero** [reply]
For face detection and the related user interface to work, multiple crops must be extracted for each face found in an image. First, this is only possible with JPEGs. Second, this requires a lot more hardware resources (CPU/memory) when extracting from larger files/originals, unless you have a team of software nerds who enjoy optimizing this particular use case without causing bugs or other issues. Considering how many users are eagerly awaiting new features (rather than such optimizations), we simply can't afford to spend endless amounts of money and time on this. Even then, I'm not sure how much benefit it would bring to users. Storage is much cheaper than development time, also considering the risks involved, such as introducing major bugs e.g. out-of-memory crashes.

**Reply by automaton82** [reply]
I appreciate the reply of "storage is cheap", but I recently spun up PhotoPrism on an LXC image (Proxmox) and pointed it at my ~500GB collection of photos (read-only).

I gave it 40GB of SSD space and it overran it before completing the initial scan which is insane to me, especially since these are just thumbnails and the originals are always there for display / use.

I am going to re-run it with the config suggestion from @djjudas21 and see. But as it is now the 'thumbnails' are not acting like thumbnails at all, they're acting like copies of the original which is just a bad design. Use the originals for everything except displaying actual thumbnails, and for actual thumbnail cache just create really small versions of it.

As a note both Windows Explorer and macOS Finder have solved this without taking many GBs of space, among others.

**Reply by lastzero** [reply]
You sound like you don't want a reply?

**Reply by martin-g-it** [reply]
Same kind of situation here like @automaton82 mentioned. 
What was your result on the suggested settings?

**Reply by automaton82** [reply]
> Same kind of situation here like @automaton82 mentioned. What was your result on the suggested settings?

Sorry I missed your comment. 1920 was still too big, I changed it to:
`PHOTOPRISM_THUMB_SIZE='1440'`
`PHOTOPRISM_JPEG_QUALITY='80'`

Using it this way I was able to get it down to ~30GB of space (10GB of savings) which is.. still not great. I may end up reducing it further, but perhaps the answer is to find an alternative software.

Even Plex, which is not a primary delivery system of photos, has this entire library available (including timeline mode) in under 5GB. How is it possible, magic?

And lastly I'm going to gripe about the fact that this file is required for all non-docker deployments and is incredibly hard to find:
`/var/lib/photoprism/.env`

I get pinged about once a month from frustrated users who can't figure out where this file is.

**Reply by lastzero** [reply]
First, you can reduce the resolution to 720. Magic. Second, non Docker deployments are third party software, so blaming us is wrong.

**Comment by Rathna-K** [comment]
but , why does it create 11 thumbnail for each image ? is there a way to restrict that number? @lastzero

**Reply by lastzero** [reply]
Yes, see config options and user guide

**Reply by graciousgrey** [reply]
--> https://docs.photoprism.app/user-guide/settings/advanced/#images

**Reply by tompayne36** [reply]
This doesn't seem to address the question "Why 11 thumbnails" and how do I adjust that...or I am missing something. Would it be bad to change the number to something like 5 if possible?

**Reply by graciousgrey** [reply]
It does ;)

- https://docs.photoprism.app/user-guide/settings/advanced/#what-files-will-be-created-by-photoprism
- https://docs.photoprism.app/user-guide/settings/advanced/#dynamic-and-static-size-limits

**Reply by tompayne36** [reply]
Ok, I guess that explains why there are 11 thumbnails. I thought there might be a way to limit to 5 (or something ) to reduce the disk usage but this makes sense now.

**Reply by lastzero** [reply]
If you reduce the resolution, there will be less. However, existing thumbnails won't get deleted by this. See our docs.

**Comment by lightmaster** [comment]
Is there a way to limit thumbnails to only small sizes for scrolling through the library of all photos, but then load the original when you click on one, like how Synology Photos does? Trying to phase out Synology from my setup, however their implementation of a photo viewer is a lot more practical, using just 290KB for 3 thumbnails vs Photoprism's 11 thumbnails totaling 1798KB for that same photo. The original photo here is just 3MB itself!

**Reply by lastzero** [reply]
See our FAQ and docs:

- https://docs.photoprism.app/getting-started/faq/#why-is-my-storage-folder-so-large-what-is-in-it

- https://docs.photoprism.app/user-guide/settings/advanced/#dynamic-and-static-size-limits

Our development preview additionally includes enhancements for PNG files so no full size JPEG is created anymore while indexing.

**Comment by doebi** [comment]
Just adding my numbers to the table:
```
470.5 GB photo collection
79.5 GB thumbnails
```

The photo collection is stored on an external hard drive, the thumbnails are stored on the internal SSD. The internal SSD has a usable capacity of 223GB, so the thumbnails are effectively eating 35% of the machines available storage.

**Comment by automaton82** [comment]
I was able to reduce mine a fair amount by just turning the thumbnails off as far as I could with this in the `.env` file:

> PHOTOPRISM_THUMB_UNCACHED='true'
> PHOTOPRISM_THUMB_FILTER='linear'
> PHOTOPRISM_THUMB_SIZE='720'
> PHOTOPRISM_THUMB_SIZE_UNCACHED='720'
> PHOTOPRISM_JPEG_SIZE='720'
> PHOTOPRISM_PNG_SIZE='720'

Despite all the dire warnings about slow performance and loading, it's been perfectly fine on all devices. Note that I have an actual server running it, not a Pi.

That said it's still taking up ~20GB of space though, but that's with ~600GB of photos and videos. 

There should be an option to just disable the thumbnails altogether. As I've opined on several times already in this thread, that is basically what Plex does in its own Photos section and nobody has a problem with it.

**Comment by lorand93** [comment]
I will start a new thread to shout my biggest concern with Photoprism. 
I really want to use photoprism but it generating 8-11 thumbnails for one photo makes my brain hurt.
As most of people do I'm hosting it on a laptop, I have plenty of power and bandwidth to serve proper thumbnails. 
I only need one thumbnail to be generated. One thumbnail for which I can control the compression and the px size limit, this is all we need for mass adoption, believe me. Have this feature in and everyone will start using photoprism.

**Reply by lorand93** [reply]
done: https://github.com/photoprism/photoprism/discussions/4327
